examples/preprocessing/plot_target_encoder.py

```diff
@@ -0,0 +1,120 @@
+# %%
+# For this example, we use the following subset of numerical and categorical
+# features in the data. The target are continuous values from 80 to 100:
+numerical_features = ["price"]
+categorical_features = [
+    "country",
+    "province",
+    "region_1",
+    "region_2",
+    "variety",
+    "winery",
+]
+target_name = "points"
+
+X = df[numerical_features + categorical_features]
+y = df[target_name]
+
+y.hist()
+
+# %%
+# Then, we split the dataset into a training and test set.
+from sklearn.model_selection import train_test_split
+
+X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=42)
+
+print(f"Samples in training set: {len(X_train)}\nSamples in test set: {len(X_test)}")
+
+# %%
+# Training and Evaluating Pipelines with Different Encoders
+# =========================================================
+# In this section, we will evaluate pipelines with
+# :class:`~sklearn.ensemble.HistGradientBoostingRegressor` with different encoding
+# strategies. First, we list out the encoders we will be using to preprocess
+# the categorical features:
+from sklearn.compose import ColumnTransformer
+from sklearn.preprocessing import OrdinalEncoder
+from sklearn.preprocessing import OneHotEncoder
+from sklearn.preprocessing import TargetEncoder
+
+categorical_preprocessors = [
+    ("drop", "drop"),
+    ("ordinal", OrdinalEncoder(handle_unknown="use_encoded_value", unknown_value=-1)),
+    (
+        "one_hot_encoder",
+        OneHotEncoder(handle_unknown="ignore", max_categories=50, sparse_output=False),
+    ),
+    ("target", TargetEncoder(target_type="continuous")),
+]
+
+# %%
+# Next, we create and fit the pipelines to the training data and evaluate them on the
+# test set:
+from sklearn.pipeline import make_pipeline
+from sklearn.ensemble import HistGradientBoostingRegressor
+from sklearn.metrics import mean_squared_error
+
+results = []
+for name, categorical_preprocessor in categorical_preprocessors:
+    preprocessor = ColumnTransformer(
+        [
+            ("nummerical", "passthrough", numerical_features),
+            ("categorical", categorical_preprocessor, categorical_features),
+        ]
+    )
+    pipe = make_pipeline(preprocessor, HistGradientBoostingRegressor(random_state=0))
+    pipe.fit(X_train, y_train)
+
+    y_pred = pipe.predict(X_test)
+    rmse = mean_squared_error(y_test, y_pred, squared=False)
+    results.append({"preprocessor": name, "root mean squared error": rmse})
+
+# %%
+# Finally, we display the results for all the encoders. When evaluating the
+# predictive performance on the test set, dropping the categories perform the
+# worst and the target encoder performs the best. The target encoding provides
+# more information about the target, which the gradient boosted model at the end
+# of the pipeline can take advantage of. In this example, we evaluate with a single
+# test split to reduce the runtime. We recommend using cross-validation to check
+# the significance of the improvement.
```

Comment:
The code wouldn't be significantly more complex if we had called `cross_validate` on the full dataset instead of calling `fit` / `predict` / `mean_squared_error` manually on a single train-test split. Don't you think we should rather do the right thing directly here?

["Show, don't tell"](https://en.wikipedia.org/wiki/Show,_don%27t_tell)

Reply:
Yea, that makes sense. I updated the PR to use `cross_validate`, but with `cv=3` to have a manageable runtime.